magefiles/installation/oauth.go

```diff
@@ -0,0 +1,120 @@
+package installation
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	retry "github.com/avast/retry-go/v4"
+	"github.com/devfile/library/pkg/util"
+	ocpOauth "github.com/openshift/api/config/v1"
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const (
+	HTPASSWD_FILE_NAME      = "ci.htpaswd"
+	DEFAULT_IDP_SECRET_NAME = "htpass-secret"
+	DEFAULT_IDP_NAME        = "ci_identity"
+	DEFAULT_OAUTH_NAME      = "cluster"
+	DEFAULT_HTPASSWD_BIN    = "htpasswd"
+)
+
+var (
+	randomOCPUserPass = util.GenerateRandomString(10)
+	randomOCPUserName = util.GenerateRandomString(6)
+	httpaswdArgs      = []string{"-c", "-B", "-b", HTPASSWD_FILE_NAME, randomOCPUserName, randomOCPUserPass}
+)
+
+// CreateOauth generate a new random admin user for testing. This functionality will be executed only in Openshift CI.
+func (i *InstallAppStudio) CreateOauth() error {
+	if os.Getenv("CI") != "true" {
+		return nil
+	}
+
+	if err := i.generateHttpaswd(); err != nil {
+		return err
+	}
+
+	if err := utils.ExecuteCommandInASpecificDirectory("oc", []string{"create", "secret", "generic", DEFAULT_IDP_SECRET_NAME, fmt.Sprintf("--from-file=htpasswd=%s/%s", i.TmpDirectory, HTPASSWD_FILE_NAME), "-n", "openshift-config"}, ""); err != nil {
+		return err
+	}
+
+	if err := i.updateOauthCluster(); err != nil {
+		return err
+	}
+	return i.loginAsNewUser()
+}
+
+// updateOauthCluster update the existing oauth object in the openshift cluster
+func (i *InstallAppStudio) updateOauthCluster() error {
+	namespacedName := types.NamespacedName{
+		Name: DEFAULT_OAUTH_NAME,
+	}
+
+	oauth := &ocpOauth.OAuth{}
+	if err := i.KubernetesClient.KubeRest().Get(context.TODO(), namespacedName, oauth); err != nil {
+		return err
+	}
+
+	updateObj := &ocpOauth.OAuth{
+		TypeMeta: v1.TypeMeta{
+			APIVersion: ocpOauth.GroupVersion.Version,
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name: DEFAULT_OAUTH_NAME,
+		},
+		Spec: ocpOauth.OAuthSpec{
+			IdentityProviders: []ocpOauth.IdentityProvider{
+				{
+					MappingMethod: ocpOauth.MappingMethodClaim,
+					Name:          DEFAULT_IDP_NAME,
+					IdentityProviderConfig: ocpOauth.IdentityProviderConfig{
+
+						Type: ocpOauth.IdentityProviderTypeHTPasswd,
+						HTPasswd: &ocpOauth.HTPasswdIdentityProvider{
+							FileData: ocpOauth.SecretNameReference{
+								Name: DEFAULT_IDP_SECRET_NAME,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	updateObj.SetResourceVersion(oauth.GetResourceVersion())
+
+	return i.KubernetesClient.KubeRest().Update(context.Background(), updateObj)
+}
+
+// generateHttpaswd create a new htpasswd file with random user and password
+func (i *InstallAppStudio) generateHttpaswd() error {
+	return utils.ExecuteCommandInASpecificDirectory(DEFAULT_HTPASSWD_BIN, httpaswdArgs, i.TmpDirectory)
```

Comment:
would it be possible to use [htpasswd golang library](https://pkg.go.dev/github.com/tg123/go-htpasswd#section-readme) instead of the binary?

Reply:
let me try generating with bcrypt the file and dont import library

Reply:
Done with: https://github.com/redhat-appstudio/e2e-tests/pull/284/files#diff-1dfcb4f8bebb12bcfa1e480cf13a37f9bf1198cd29b988b73d3fb3398c1092b8R38